lib/synced/strategies/full.rb

```diff
@@ -147,7 +147,7 @@ def api
         closest = [@scope, @scope.class, @model_class].find do |object|
                     object.respond_to?(:api)
                   end
-        closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
+        @api ||= closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
```

Comment:
Do we have a test for this case? That if I call this method from an object twice, on second call we will receive exactly the same object as on the first one.